vendor/github.com/openshift/library-go/pkg/operator/resource/resourceapply/prometheus.go

```diff
@@ -0,0 +1,71 @@
+package resourceapply
+
+import (
+	"fmt"
+
+	"github.com/ghodss/yaml"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+)
+
+var serviceMonitorGVR = schema.GroupVersionResource{Group: "monitoring.coreos.com", Version: "v1", Resource: "servicemonitors"}
+
+// ApplyServiceMonitor applies the Prometheus service monitor.
+func ApplyServiceMonitor(client dynamic.Interface, recorder events.Recorder, serviceMonitorBytes []byte) (bool, error) {
+	monitorJSON, err := yaml.YAMLToJSON(serviceMonitorBytes)
+	if err != nil {
+		panic(err)
+	}
+	monitorObj, err := runtime.Decode(unstructured.UnstructuredJSONScheme, monitorJSON)
+	if err != nil {
+		panic(err)
+	}
+	required, ok := monitorObj.(*unstructured.Unstructured)
+	if !ok {
+		panic(fmt.Sprintf("unexpected object in %t", monitorObj))
+	}
+
+	existing, err := client.Resource(serviceMonitorGVR).Get(required.GetName(), metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		_, createErr := client.Resource(serviceMonitorGVR).Create(required)
+		if createErr != nil {
+			recorder.Warningf("ServiceMonitorCreateFailed", "Failed to create ServiceMonitor.monitoring.coreos.com/v1: %v", createErr)
+			return true, createErr
```

Comment:
not `return false, createErr`?

Reply:
no, we return true in this case from other apply func... i guess it indicates that we detected a change (not found) but failed to react to it (error)